examples/src/main/scala/za/co/absa/fadb/examples/enceladus/DatasetSchema.scala

```diff
@@ -60,72 +59,63 @@ object DatasetSchema {
+    with SlickPgFunctionWithStatusSupport[SchemaInput, Long]
+    with UserDefinedStatusHandling {
 
-
-    override protected def sqlToCallFunction(values: SchemaInput): SQLActionBuilder = {
+    override protected def sql(values: SchemaInput): SQLActionBuilder = {
       sql"""SELECT A.status, A.status_text, A.id_schema_version
-             FROM #$functionName(${values.schemaName}, ${values.schemaVersion}, ${values.schemaDescription},
-                ${values.fields}::JSONB, ${values.userName}
-             ) A;"""
+            FROM #$functionName(${values.schemaName}, ${values.schemaVersion}, ${values.schemaDescription},
+              ${values.fields}::JSONB, ${values.userName}
+            ) A;"""
     }
 
-    override protected def resultConverter: GetResult[Long] = {
-      val gr:GetResult[Long] = GetResult(r => {
-        val status: Int = r.<<
-        val statusText: String = r.<<
-        if (status != 201) throw new StatusException(status, statusText)
-        r.<<
-      })
-      gr
-    }
+    override protected def slickConverter: GetResult[Long] = GetResult.GetLong
+
+    override def OKStatuses: Set[Integer] = Set(201)
+
   }
 
-  final class GetSchema(implicit schema: DBSchema[ExecutorEngineType])
-    extends DBUniqueFunction[ExecutorEngineType, (String, Option[Int]), Schema](schema)
-    with SlickPgFunction[(String, Option[Int]), Schema] {
+  final class GetSchema(implicit override val schema: DBSchema, override val dbEngine: SlickPgEngine)
+    extends DBUniqueFunction[(String, Option[Int]), Schema, SlickPgEngine]
+    with SlickPgFunctionWithStatusSupport[(String, Option[Int]), Schema]
+    with UserDefinedStatusHandling {
 
     def apply(id: Long): Future[Schema] = {
       val sql =
         sql"""SELECT A.*
              FROM #$functionName($id) A;"""
 
-      schema.unique(makeQueryFunction(sql)(resultConverter))
+      val slickQuery = new dBEngine.QueryType(sql, slickConverter)
+      dBEngine.unique[Schema](slickQuery)
     }
 
-    override protected def sqlToCallFunction(values: (String, Option[Int])): SQLActionBuilder = {
+    override protected def sql(values: (String, Option[Int])): SQLActionBuilder = {
       sql"""SELECT A.*
-              FROM #$functionName(${values._1}, ${values._2}) A;"""
+            FROM #$functionName(${values._1}, ${values._2}) A;"""
+    }
+
+    override protected val slickConverter: GetResult[Schema] = GetResult{r =>
+      Schema(r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<, r.<<)
     }
 
-    override protected def resultConverter: GetResult[Schema] = DatasetSchema.GetSchemaImplicit
+    override val OKStatuses: Set[Integer] = Set(200)
   }
 
-  final class ListSchemas(implicit schema: DBSchema[ExecutorEngineType])
-    extends DBSeqFunction[ExecutorEngineType, Boolean, SchemaHeader](schema)
+  final class List(implicit override val schema: DBSchema, override val dbEngine: SlickPgEngine)
```

Comment:
I know that this is just an example but maybe it can be further improved?

Why `ListSchemas` -> `List`? `List` is quite general and the name can be both verb (list/enumerate something) or a noun (list as a data type). I would prefer `ListSchemas` or `ListSchemaHeaders` if you want

Reply:
Yeeaah.
Well the examples were created for Enceladus migration to Postgres. Then the function name in the DB model in that PR were changed. But before reflecting it here, the whole migration was shelved.
The examples confidently show some interesting methods how the `DBFunction` successors can operate:
* having or not having _status_ support
* defining additional `apply` function
* overriding the existing one
So they might serve some benefit until #9 is delivered.

But I wanted to test it against the Enceladus DB model, just to verify, the refactoring works in practice, not just compilation. That's why the rename... Might be rolled back. 🤔

Reply:
okay I kind of thought so. Well, since there is this #9, which will change everything, I don't mind keeping as is. Maybe, if you are up for it, add some brief code doc, maybe 1-2 sentences can be enough. But up to you - I at least wanted to have some understanding and now I have :)

Reply:
_The examples confidently show..._ -> The examples coincidentally show...